backends/generators/generator.py

```
@@ -17,6 +17,102 @@ def check_requirement(req, exts):
+                    if "-" in location:
+                        high = int(location.split("-")[0])
+                    else:
+                        high = int(location)
+                    valid_locations.append(high)
+                except (ValueError, IndexError):
+                    continue  # Skip invalid location formats
+            elif isinstance(var_data["location"], int):
+                try:
+                    valid_locations.append(var_data["location"])
+                except (ValueError, IndexError):
+                    continue  # Skip invalid location formats
+            else:
+                continue  # Skip invalid location formats
+
+    if not valid_locations:
+        return None
+
+    max_bit = max(valid_locations)
+
+    # Set instruction width based on maximum bit position
+    width = max_bit + 1
+    match_bits = ["-"] * width
+
+    # Populate match string with opcode bits
+    for field_data in opcodes.values():
+        if (
+            isinstance(field_data, dict)
+            and "location" in field_data
+            and "value" in field_data
+            and isinstance(field_data["location"], str)
+            and isinstance(field_data["value"], int)
+        ):
+
+            try:
+                location = field_data["location"]
+                if "-" in location:
+                    high, low = map(int, location.split("-"))
+                else:
+                    high = low = int(location)
+
+                if high < low or high >= width:
+                    continue  # Skip invalid bit ranges
+
+                binary_value = format(field_data["value"], f"0{high - low + 1}b")
+                for i, bit in enumerate(binary_value):
+                    pos = high - i
+                    if 0 <= pos < width:
+                        match_bits[width - 1 - pos] = bit
+            except (ValueError, IndexError):
+                continue  # Skip invalid field data
```

I think this also is a case where we do not want to ignore errors.